### PR TITLE
move config.Consensus to consensus package

### LINKED ISSUE #3504: move config.Consensus to consensus package
### Description of the code refactor
it contains `RollDPoS` struct, so need to handle that as well

### Why is this needed
- [] better/less memory usage
- [] reduce running time
- [] clean (easy-to-understand) data structure or algorithm
- [] remove dead code
- [x] better code design or structure (like reduce complexity)
- [x] reduce/decouple dependencies between code modules
- [] increase test coverage
- [x] better code readability

### PR DESCRIPTION
**Pull request by millken**
# Description

This PR contains changes for two issues

Fixes #3504 #3506

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by millken** [comment]
Config moved to a separate folder is to prevent circular imports

**Comment by Liuhaai** [comment]
move the default config of consensus to `consensus` pkg
and import the value in `config.go` like 
https://github.com/iotexproject/iotex-core/blob/d6eaa2efddfc6691b0eebcb5a0d2a4c68f444597/config/config.go#L185

File: consensus/consensusfsm/consensus_ttl.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ import (
 
 	"github.com/iotexproject/iotex-core/blockchain/genesis"
 	"github.com/iotexproject/iotex-core/config"
```
Comment (dustinxie) [inline consensus/consensusfsm/consensus_ttl.go:13]: remove this dependency, by breaking inputs to `NewConsensusConfig` at L42

**Comment by millken** [comment]
> 2. `config.go` should be put under `./consensus` folder.

can not put under `./consensus` folder. there will be a error `Import cycle not allowed`

**Comment by dustinxie** [comment]
> put this on a brief hold, #3511 should solve the import cycle, try again after 3511 merges

3511 has merged, pls try again

**Comment by raullenchai** [comment]
> > put this on a brief hold, #3511 should solve the import cycle, try again after 3511 merges
> 
> 3511 has merged, pls try again

@millken pls try again

**Pull request closed without merging**